common/src/uds_fe_programming.rs

```diff
@@ -144,22 +148,38 @@ impl UdsFeProgrammingFlow {
             let fuse_controller_base_addr = soc_ifc.fuse_controller_base_addr();
             let otp_ctrl = DmaOtpCtrl::new(AxiAddr::from(fuse_controller_base_addr), dma);
             let seed_dest_address = self.get_dest_address(soc_ifc);
+            let dai_idle_bit_num = soc_ifc.otp_dai_idle_bit_num();
+            let direct_access_cmd_reg_addr =
+                fuse_controller_base_addr + soc_ifc.otp_direct_access_cmd_reg_offset() as u64;
```

Comment:
whats the point of retrieving the CMD reg offset from the generic straps, but assumming the rest of the registers (ADDR, WDATA_0, WDATA_1) will be at the correct relative offset? 

If you know the base address of the fuse controller, which it looks like you do on 149, shouldn't that be enough?

Reply:
The registers that are optional (really, variable) is the error registers, which depend on the number of partitions. The integrator can customize the number of partitions, which changes the location of the DAI IDLE bit and the CMD register.